packages/compiler/src/render3/view/styling_builder.ts

```diff
@@ -377,22 +379,32 @@ export class StylingBuilder {
     // be evaluated (the AST visit call) during creation time so that any
     // pipes can be picked up in time before the template is built
     const mapValue = stylingInput.value.visit(valueConverter);
+    let reference: o.ExternalReference;
     if (mapValue instanceof Interpolation) {
+      if (!isClassBased) {
+        // We only support interpolations inside class bindings.
+        error('Unexpected interpolation');
```

Comment:
Can we add some more info in the error? Like

```suggestion
    error('Unexpected interpolation. Interpolated values are not supported in the `style` property.');
```